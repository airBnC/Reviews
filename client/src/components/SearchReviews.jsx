import React from 'react';

const SearchReviews = props => (
  <div>
    <h6> Search reviews </h6>
  </div>
);

export default SearchReviews;