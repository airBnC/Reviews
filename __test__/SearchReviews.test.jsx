import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import SearchReviews from '../client/src/components/SearchReviews';

configure({ adapter: new Adapter() });

const dummyCurrentSearch = 'Diablo II';

describe('SearchReviews', () => {
  it('should render correctly', () => {
    const output = shallow(<SearchReviews currentSearch={dummyCurrentSearch} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

// Notes to self after experimenting with tests of this component:
  /*
    - No point testing if there are default props.  Covered by PropTypes library.
    - No point testing default prop values.  Covered by PropTypes library.
    - No point testing whether handleSearchChange executes on change.
      --That would be testing React.
      --If you say get rid of that prop altogether, the snapshot catches it.
      --Same goes for handleSearchSubmit.
    - No point testing whether handleSearchChange is functioning correctly.
      --You can't, because this is unit testing, and that function is passed down.
      --Same goes for handleSearchSubmit.
    - No point testing styling.
      --That would be caught by the snapshot test.
    - There's actually no point passing in dummy data.  Covered by PropTypes library.
      --You're not going to go back and delete it all though.  (Maybe later!)
  */
