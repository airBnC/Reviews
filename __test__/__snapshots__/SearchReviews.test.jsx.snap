// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchReviews should render correctly 1`] = `
<styled.span>
  <styled.input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search reviews"
    type="text"
    value="Diablo II"
  />
</styled.span>
`;
